### PR TITLE
fix default key encoding

### PR DESCRIPTION
**Pull request by juliangruber**
The db i passed to hyperdrive/hypercore had `{keyEncoding: bytewise}`, so this was causing bugs.

The comment I removed said "binary" encoding, but I the keys written were all strings so I used `utf8` for simplicity.

### CODE REVIEW

File: test/helpers/create.js [hunk 1]
```diff
@@ -2,5 +2,5 @@ var hypercore = require('../../')
 var memdb = require('memdb')
 
 module.exports = function create () {
-  return hypercore(memdb())
+  return hypercore(memdb({keyEncoding: 'json'}))
```
Comment (mafintosh) [inline test/helpers/create.js]: i'm curious why this is needed?

Comment (juliangruber) [inline test/helpers/create.js]: to make the tests prove that you can supply any key encoding. should i add a comment?

Comment (mafintosh) [inline test/helpers/create.js]: ah cool! yea that'd be great.

Comment (juliangruber) [inline test/helpers/create.js]: done!

**Pull request merged**